src/libraries/System.Console/src/System/IO/ConsoleStream.cs

```diff
@@ -28,6 +30,32 @@ public override void Write(byte[] buffer, int offset, int count)
 
         public override void WriteByte(byte value) => Write(new ReadOnlySpan<byte>(in value));
 
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            ValidateWrite(buffer, offset, count);
```

Comment:
We typically prefer to do argument validation before cancellation checks.

Reply:
Oh, I took this order from the base Stream.WriteAsync:

https://github.com/dotnet/runtime/blob/cae954add0f01c85de7fef8cc41c2a11d1c920e1/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L737-L742

Reply:
Yeah, we're very inconsistent on this front.

You can leave it, but the reason we prefer argument validation first is:
- Those are the only exceptions we allow / prefer to emerge synchronously, and so we want to do them first and then switch over into a mode where all exceptions go out through the task.
- Argument exceptions are typically deterministic usage errors (not always, but typically), whereas cancellation is transient.  We want the deterministic things to happen first and not become non-deterministic due to cancellation sometimes being requested prior to the method being invoked.

Reply:
I can definitely switch these around though, if that is the usual order.